### PR TITLE
Refactor Ai class and difficulties using the "Strategy pattern"

### PR DESCRIPTION
**Pull request by GeorgeSmith-Sweeper**
- Created easy, medium, and hard difficulties classes that implement IStrategy interface.
- Modified setAiDifficulties in Config to create difficultly instances.
- Creates interfaces package.
- Moved ai tests to their respective difficulties.

### CODE REVIEW

File: src/test/java/com/EighthLight/app/TestAiDifficulties.java [hunk 1]
```diff
@@ -8,6 +8,13 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class TestAiDifficulties {
+    private String aiSymbol = "X";
+    private String humanSymbol = "O";
+    private Board board3x3 = new Board(3);
+    private Board board4x4 = new Board(4);
+    private EasyDifficulty easy = new EasyDifficulty(aiSymbol);
+    private HardDifficulty hard = new HardDifficulty(aiSymbol, humanSymbol);
```
Comment (GeorgeSmith-Sweeper) [inline src/test/java/com/EighthLight/app/TestAiDifficulties.java:17]: Create fields to reduce duplication in tests

File: src/main/java/com/EighthLight/app/IStrategy.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 package com.EighthLight.app;
 
 public interface IStrategy {
-    void markBoard(IBoard board, String aiSymbol, String humanSymbol);
+    void markBoard(IBoard board);
```
Comment (GeorgeSmith-Sweeper) [inline src/main/java/com/EighthLight/app/IStrategy.java]: IStrategy interface provides the "contract" for all difficultly methods to follow. All difficulties must have a markBoard method that takes a board. The constructors in each difficulty provide information unique to that instance.

File: src/test/java/com/EighthLight/app/TestGame.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package com.EighthLight.app;
 
+import com.EighthLight.app.Interfaces.IPlayer;
```
Comment (nicole-a-tesla) [inline src/test/java/com/EighthLight/app/TestGame.java:3]: Unused import?

Comment (GeorgeSmith-Sweeper) [reply src/test/java/com/EighthLight/app/TestGame.java:3]: used to set the type when creating an ArrayList of Players eg. ArrayList(IPlayer) players = new ArrayList();

**Pull request merged**